lib/improver/tests/utilities/test_spatial.py

```diff
@@ -151,82 +153,114 @@ def setUp(self):
         self.additional_data = additional_data
 
 
+class Test_calculate_grid_spacing(IrisTest):
+    """Test the calculate_grid_spacing function"""
+
+    def setUp(self):
+        """Set up an equal area cube"""
+        self.cube = set_up_variable_cube(
+            np.ones((5, 5), dtype=np.float32), spatial_grid='equalarea')
+        self.spacing = 200000.0
+
+    def test_basic(self):
+        """Test correct answer is returned from an equal area grid"""
+        result = calculate_grid_spacing(self.cube)
+        self.assertAlmostEqual(result, self.spacing)
+
+    def test_units(self):
+        """Test correct answer is returned for coordinates in km"""
+        for axis in ['x', 'y']:
+            self.cube.coord(axis=axis).convert_units('km')
+        result = calculate_grid_spacing(self.cube)
+        self.assertAlmostEqual(result, self.spacing)
+        for axis in ['x', 'y']:
+            self.assertEqual(self.cube.coord(axis=axis).units, 'km')
+
+    def test_axis_keyword(self):
+        """Test using the other axis"""
+        self.cube.coord(axis='y').points = 2*(self.cube.coord(axis='y').points)
+        result = calculate_grid_spacing(self.cube, axis='y')
+        self.assertAlmostEqual(result, 2*self.spacing)
+
+    def test_cube_not_equal_area(self):
+        """Test ValueError if cube is not equal area"""
+        cube = set_up_variable_cube(np.ones((5, 5), dtype=np.float32))
+        with self.assertRaises(ValueError):
```

Comment:
Can you please add the message that is raised to the test condition. This avoids some other unexpected ValueError being treated as a expected behaviour.